src/taskgraph/util/dict_helpers.py

```diff
@@ -0,0 +1,10 @@
+def deep_get(dict_, field):
```

Comment:
As per your other PR, I think this is similar to the functions in `util/templates.py`. I agree the name isn't the greatest, but imo better to keep similar functions grouped together.